src/groovy/org/transmartproject/db/util/SimpleScrollableResults.groovy

```diff
@@ -0,0 +1,55 @@
+package org.transmartproject.db.util
+
+import org.hibernate.ScrollableResults
+
+class SimpleScrollableResults<T> implements Closeable, Iterable<T> {
+
+    private final SimpleScrollableResultsIterator scrollableResultsIterator
+
+    SimpleScrollableResults(ScrollableResults scrollableResults) {
+        this.scrollableResultsIterator = new SimpleScrollableResultsIterator()
+        this.scrollableResultsIterator.scrollableResults = scrollableResults
+    }
+
+    @Override
+    Iterator<T> iterator() {
+        scrollableResultsIterator
+    }
+
+    @Override
+    void close() throws IOException {
+        this.scrollableResultsIterator.scrollableResults.close()
+    }
+
+    class SimpleScrollableResultsIterator implements Iterator<T> {
+
+        private ScrollableResults scrollableResults
+        private Boolean hasNext = null
+
+        @Override
+        boolean hasNext() {
+            if (hasNext == null) {
+                hasNext = scrollableResults.next()
+            } else {
+                hasNext
+            }
+        }
+
+        @Override
+        T next() {
+            if (hasNext()) {
+                hasNext = null
+                scrollableResults.get(0)
+            } else {
+                throw new NoSuchElementException()
+            }
+        }
+
+        @Override
+        void remove() {
+            throw new UnsupportedOperationException()
+        }
+
+    }
+
```

Comment:
Can you override `finalize()` like in `CollectingTabularResult`? It really helps detecting leaks.